dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraph.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.dependencies;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.aether.artifact.Artifact;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.SetMultimap;
+import com.google.common.collect.TreeMultimap;
+
+/**
+ * A representation of the complete non-cyclic transitive dependency tree of 
+ * a Maven artifact.
+ * 
+ * The representation is unusual. Imagine we perform a breadth first search 
```

Comment:
I'm unclear on why the representation is unusual.

Reply:
Perhaps it's not. This represents a tree as a list of every path from the root to each node. Is that a common thing to do?

Reply:
done